File: audit_samples/vulnerable/sqli.go
package main

import (
	"database/sql"
	"fmt"
)

func getUser(db *sql.DB, id string) {
	// VULNERABLE: Direct interpolation via fmt.Sprintf
	query := fmt.Sprintf("SELECT * FROM users WHERE id = %s", id)
	db.Query(query)
}

func main() {
	fmt.Println("Vulnerable Go code")
}

File: audit_samples/vulnerable/sqli.ts
// Vulnerable SQL Injection
const userId = "123";
const query1 = "SELECT * FROM users WHERE id = " + userId;
const query2 = `SELECT * FROM users WHERE id = ${userId}`;

function getUser(id: string) {
    const sql = "SELECT * FROM accounts WHERE acc_id = " + id;
    return db.execute(sql);
}
